packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -2316,14 +2316,28 @@ Finder _opacityAboveLabel(String text) {
 }
 
 // Only valid when labelType != all.
-double _labelOpacity(WidgetTester tester, String text) {
-  final Opacity opacityWidget = tester.widget<Opacity>(
-    find.ancestor(
-      of: find.text(text),
-      matching: find.byType(Opacity),
-    ),
-  );
-  return opacityWidget.opacity;
+double? _labelOpacity(WidgetTester tester, String text) {
+  // We search for both Opacity and FadeTransition since in some
+  // cases opacity is animated, in other it's not.
+  try {
```

Comment:
please don't use try/catch for control flow.

Reply:
we both use Opacity and FadeTransition - if former was not found in tree, it should pick the latter one. i am not sure what i should do about this?

Reply:
fixed it